redfish-core/lib/assembly.hpp

```diff
@@ -51,6 +50,52 @@ inline void
         tempyArray.at(assemblyIndex)["Status"]["Health"] = "OK";
         tempyArray.at(assemblyIndex)["Status"]["State"] = "Enabled";
 
+        // Handle special case for tod_battery assembly OEM ReadyToRemove property
+        // NOTE: The following method for the special case of the tod_battery
+        // ReadyToRemove property only works when there is only ONE adcsensor
+        // handled by the adcsensor application.
+        if (sdbusplus::message::object_path(assembly).filename() == "tod_battery")
+        {
+            tempyArray.at(assemblyIndex)["Oem"]["OpenBMC"]["@odata.type"] =
+                                                "#OemAssembly.v1_0_0.Assembly";
+
+            crow::connections::systemBus->async_method_call(
+                [aResp, assemblyIndex](
+                    const boost::system::error_code ec,
+                    const std::vector<std::pair<std::string, std::vector<std::string>>>&
+                          object) {
+                    if (ec)
+                    {
+                        if (!((std::strcmp(ec.category().name(), "generic") == 0) 
+                                                            && ( ec.value() == 5))) 
+                        {
+                            BMCWEB_LOG_DEBUG << "DBUS response error";
+                            messages::internalError(aResp->res);
+                            return;
+                        }
+                    }
+
+                    nlohmann::json& assemblyArray = 
+                                      aResp->res.jsonValue["Assemblies"];
+
+                    if (object.size() == 0)
```

Comment:
If "/xyz/openbmc_project/sensors/voltage/Battery_Voltage" is not found on DBUS by getObject, then the size of object is 0.  See the example: https://github.com/ibm-openbmc/bmcweb/blob/893571ce6b1a1463aac73508fa942716aad36bb9/redfish-core/lib/oem/ibm/lamp_test.hpp#L42
Currently ADCSensor is supporting only one ADC.  When it starts, it will create the DBUS object for the Battery Voltage.